drivers/sensor/vl53l1x/vl53l1.c

```diff
@@ -0,0 +1,479 @@
+/* vl53l1.c - Driver for ST VL53L1X time of flight sensor */
+
+/*
+ * Copyright (c) 2023 Prosaris SOlutions Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_vl53l1x
+
+#include <errno.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+
+#include "vl53l1_api.h"
+#include "vl53l1_platform.h"
+
+LOG_MODULE_REGISTER(VL53L1X, CONFIG_SENSOR_LOG_LEVEL);
+
+struct vl53l1x_config {
+	struct i2c_dt_spec i2c;
+#ifdef CONFIG_VL53L1X_XSHUT
+	struct gpio_dt_spec xshut;
+#endif
+#ifdef CONFIG_VL53L1X_INTERRUPT_MODE
+	struct gpio_dt_spec gpio1;
+#endif
+};
+
+struct vl53l1x_data {
+	VL53L1_Dev_t vl53l1x;
+	VL53L1_RangingMeasurementData_t data;
+	VL53L1_DistanceModes distance_mode;
+#ifdef CONFIG_VL53L1X_INTERRUPT_MODE
+	struct gpio_callback gpio_cb;
+	struct k_work work;
+	const struct device *dev;
+#endif
+};
+
+static VL53L1_Error vl53l1x_read_sensor(struct vl53l1x_data *drv_data)
+{
+	VL53L1_Error ret;
+
+	ret = VL53L1_GetRangingMeasurementData(&drv_data->vl53l1x, &drv_data->data);
```

Comment:
I can't seem to find where this is implemented, can you point me in the right direction?

Reply:
Hi @yperess. Thanks for taking a look. This is part of a library by ST Microelectronics which was converted to a Zephyr module by some people before my time. It is related to how Zephyr uses external modules using multiple repositories and west as the manager of all ofthat. 

It exists in `modules\hal\st\sensor\vl53l1x` when imported (using the west manifest) when this driver is included in the build of an application (see [here](https://github.com/zephyrproject-rtos/hal_st/blob/master/sensor/vl53l1x/)). You can find the definition of that particular function [here](https://github.com/zephyrproject-rtos/hal_st/blob/master/sensor/vl53l1x/api/core/src/vl53l1_api.c#:~:text=VL53L1_Error%20VL53L1_GetRangingMeasurementData).

It is imported by Zephyr in the west.yml [here](https://github.com/markCwatson/zephyr/blob/main/west.yml#:~:text=path%3A%20modules/hal/st).

I hope all of this makes sense.

@stephanosio @erwango @MaureenHelm @nashif Correct me if my description is wrong in any way.